Source/cmake/FindGLibCompileResources.cmake

```
@@ -0,0 +1,84 @@
+    OUTPUT_VARIABLE glib_compile_resources_version
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "Found glib-compile-resources (required): "
+    ${GLIB_COMPILE_RESOURCES_EXE}
+    " (${glib_compile_resources_version})")
+if (${glib_compile_resources_version} VERSION_LESS 2.77)
+    set(glib_compile_resources_has_depfile_bug TRUE)
+    message(STATUS "  Needs workaround for depfiles (will be used automatically)")
+else ()
+    set(glib_compile_resources_has_depfile_bug FALSE)
+    message(STATUS "  Doesn't need workarounds")
+endif ()
+
+function(GLIB_COMPILE_RESOURCES)
+    set(zeroArgKeywords "")
+    set(oneArgKeywords OUTPUT SOURCE_XML DEPFILE)
+    set(manyArgsKeywords RESOURCE_DIRS)
+    cmake_parse_arguments(
+        PARSE_ARGV 0 ARG
+        "${zeroArgKeywords}" "${oneArgKeywords}" "${manyArgsKeywords}"
+    )
+
+    file(RELATIVE_PATH rel_output       ${CMAKE_BINARY_DIR} ${ARG_OUTPUT})
+    file(RELATIVE_PATH rel_source_xml   ${CMAKE_BINARY_DIR} ${ARG_SOURCE_XML})
+    set(resource_dir_args "")
+    foreach (abs_resource_dir IN LISTS ARG_RESOURCE_DIRS)
+        file(RELATIVE_PATH rel_resource_dir ${CMAKE_BINARY_DIR} ${abs_resource_dir})
+        list(APPEND resource_dir_args --sourcedir=${rel_resource_dir})
+    endforeach ()
+
+    set(additional_cmd_line)
+    if (${glib_compile_resources_has_depfile_bug})
+        # Workaround for older versions of glib-compile-resources lacking this fix:
+        # https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3460
+        #
+        # Affected versions produce broken depfiles that look like this:
+        #   foo.xml: resource1 resource2
+        # But depfiles should look like this:
+        #   foo.c: foo.xml resource1 resource2
+
+        # CMake deals very poorly with dollar signs inside add_custom_command():
+        # https://gitlab.kitware.com/cmake/cmake/-/issues/18062
+        # To avoid that issue, save the short perl script as a file instead of
+        # running it with `perl -e`.
+        set(perl_code [==[
+            # Workaround for https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3460
+            # Produced by FindGLibCompileResources.cmake for glib-compile-commands < 2.77
+            BEGIN {
+                ($old_xml, $new_target) = splice(@ARGV, 0, 2);
+            }
+            s{\Q$old_xml:}{$new_target: $old_xml}
+        ]==])
+        file(WRITE ${CMAKE_BINARY_DIR}/fix_glib_compile_resources_depfile.pl ${perl_code})
+
+        set(additional_cmd_line &&
+            perl -pi ${CMAKE_BINARY_DIR}/fix_glib_compile_resources_depfile.pl
+            ${rel_source_xml} ${rel_output} ${ARG_DEPFILE})
+    endif ()
+
+    add_custom_command(
+        OUTPUT  ${ARG_OUTPUT} ${ARG_DEPFILE}
+        DEPENDS ${ARG_SOURCE_XML}
+        DEPFILE ${ARG_DEPFILE}
+        COMMAND ${GLIB_COMPILE_RESOURCES_EXE}
+                --generate
+                --target=${rel_output}
+                --dependency-file=${ARG_DEPFILE}
```

To accomodate for the removed `DEPFILE` keyword argument, derive the path from `ARG_OUTPUT`, for example:

```suggestion
        OUTPUT  ${ARG_OUTPUT} ${ARG_OUTPUT}.deps
        DEPENDS ${ARG_SOURCE_XML}
        DEPFILE ${ARG_OUTPUT}.deps
        COMMAND ${GLIB_COMPILE_RESOURCES_EXE}
                --generate
                --target=${rel_output}
                --dependency-file=${ARG_OUTPUT}.deps
```